python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java

```diff
@@ -993,6 +994,84 @@ void try_except_list_attributes() {
 
   }
 
+  @Test
+  void inferTypeForQualifiedExpression() {
+    var root = inferTypes("""
+      class A:
+        def foo():
+           ...
+      def f():
+        a = A()
+        a.foo()
+      """);
+
+    var qualifiedExpression = TreeUtils.firstChild(root.statements().statements().get(1), QualifiedExpression.class::isInstance)
+      .map(QualifiedExpression.class::cast)
+      .get();
+
+    Assertions.assertThat(qualifiedExpression)
+      .isNotNull()
+      .extracting(QualifiedExpression::typeV2)
+      .isNotNull();
+
+    var qualifiedExpressionType = qualifiedExpression.typeV2();
+    Assertions.assertThat(qualifiedExpressionType)
+      .isInstanceOf(FunctionType.class)
+      .extracting(PythonType::name)
+      .isEqualTo("foo");
+  }
+
+
+  @Test
+  void inferBuiltinsTypeForQualifiedExpression() {
+    var root = inferTypes("""
+      a = [42]
+      a.append(1)
+      """);
+
+    var qualifiedExpression = TreeUtils.firstChild(root.statements().statements().get(1), QualifiedExpression.class::isInstance)
+      .map(QualifiedExpression.class::cast)
+      .get();
+
+    Assertions.assertThat(qualifiedExpression)
+      .isNotNull()
+      .extracting(QualifiedExpression::typeV2)
+      .isNotNull();
+
+    var qualifiedExpressionType = qualifiedExpression.typeV2();
+    Assertions.assertThat(qualifiedExpressionType)
+      .isInstanceOf(FunctionType.class)
+      .extracting(PythonType::name)
+      .isEqualTo("append");
+  }
+
+  @Test
+  @Disabled("Attribute types resolving")
+  void inferBuiltinsAttributeTypeForQualifiedExpression() {
+    var root = inferTypes("""
+      def f():
+        e = OSError()
+        e.errno
+      """);
+
+    var qualifiedExpression = TreeUtils.firstChild(root.statements().statements().get(0), QualifiedExpression.class::isInstance)
+      .map(QualifiedExpression.class::cast)
+      .get();
+
+    Assertions.assertThat(qualifiedExpression)
+      .isNotNull()
+      .extracting(QualifiedExpression::typeV2)
+      .isNotNull();
+
+    var qualifiedExpressionType = qualifiedExpression.typeV2();
+    Assertions.assertThat(qualifiedExpressionType)
+      .isInstanceOf(ObjectType.class)
+      .extracting(ObjectType.class::cast)
+      .extracting(ObjectType::type)
+      .extracting(PythonType::name)
+      .isEqualTo("int");
+  }
+
```

Comment:
Another case that would be interesting to test is when qualified expressions are assigned to a variable.
```
class A:
  def foo(self): ...

a = A()
b = a.foo
b() # is the type of b the same as the type of a.foo?
```